prov/util/src/util_mr_cache.c

```diff
@@ -135,6 +136,18 @@ static void util_mr_uncache_entry(struct ofi_mr_cache *cache,
 	}
 }
 
+static bool ofi_mr_cache_entry_usable(struct ofi_mr_entry *entry,
+				      const struct fi_mr_attr *attr,
+				      struct iovec *iov)
+{
+#ifdef HAVE_LIBCUDA
+	return (ofi_cuda_valid_reg(entry, iov)
```

Comment:
We should be able to store whether a cached entry was a cuda buffer or not  at registration time, and skip the cuda calls for non-cuda buffers.

Reply:
Yeah, this would have been easier to do from the EFA provider, since the entry's data has this information. The cache entry could stash the reg attributes too, which is richer than having a cuda flag.

In https://github.com/ofiwg/libfabric/issues/5789#issuecomment-608640709, I was also suggesting we could extend fi_mr_attr to give us this buffer ID for FI_HMEM_CUDA. We could then just skip the this in the create path entirely (if we go down this path).